drivers/afe/ad413x/ad413x.c

```diff
@@ -0,0 +1,816 @@
+ * @param reg_nb - Number of Configuration Register
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ad413x_set_ref(struct ad413x_dev *dev, enum ad413x_ref_sel ref,
+		       uint8_t reg_nb)
+{
+	int32_t ret;
+
+	ret = ad413x_spi_int_reg_write_msk(dev,
+					   AD413X_REG_CONFIG(reg_nb),
+					   AD413X_REF_SEL_N(ref),
+					   AD413X_REF_SEL_N(0xF));
+	if (ret)
+		return ret;
+
+	dev->preset[reg_nb].ref_sel = ref;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * Select filter from filter register.
+ *
+ * @param dev  	 - The device structure.
+ * @param filter - The gain value.
+ * 		 Accepted values: AD413X_SYNC4_STANDALONE
+ *					AD413X_SYNC4_SYNC1
+ *					AD413X_SYNC3_STANDALONE
+ *					AD413X_SYNC3_REJ60
+ *					AD413X_SYNC3_SYNC1
+ *					AD413X_SYNC3_PF1
+ *					AD413X_SYNC3_PF2
+ *					AD413X_SYNC3_PF3
+ *					AD413X_SYNC3_PF4
+ * @param reg_nb - Number of Configuration Register
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ad413x_set_filter(struct ad413x_dev *dev, enum ad413x_filter filter,
+			  uint8_t reg_nb)
+{
+	int32_t ret;
+
+	ret = ad413x_spi_int_reg_write_msk(dev,
+					   AD413X_REG_FILTER(reg_nb),
+					   AD413X_FILTER_MODE_N(filter),
+					   AD413X_FILTER_MODE_N(0xF));
+	if (ret)
+		return ret;
+
+	dev->preset[reg_nb].filter = filter;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * Select preset for adc channel.
+ *
+ * @param dev  		- The device structure.
+ * @param ch_nb  	- The channel number.
+ * @param preset_nb - The preset number.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ad413x_set_ch_preset(struct ad413x_dev *dev, uint8_t ch_nb,
+			     uint8_t preset_nb)
+{
+	int32_t ret;
+
+	ret = ad413x_spi_int_reg_write_msk(dev,
+					   AD413X_REG_CHN(ch_nb),
+					   AD413X_SETUP_M(preset_nb),
+					   AD413X_SETUP_M(0xF));
+	if (ret)
+		return ret;
+
+	dev->ch[ch_nb].preset = preset_nb;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * Enable/disable channel.
+ *
+ * @param dev  		- The device structure.
+ * @param ch_nb  	- The channel number.
+ * @param enable 	- Channel status.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ad413x_ch_en(struct ad413x_dev *dev, uint8_t ch_nb, uint8_t enable)
+{
+	int32_t ret;
+
+	if (enable) {
+		ret = ad413x_spi_int_reg_write_msk(dev,
+						   AD413X_REG_CHN(ch_nb),
+						   AD413X_ENABLE_M,
+						   AD413X_ENABLE_M);
+		if (ret)
```

Comment:
this can be condensed to:
```
if (enable)
      ret = ad413x_spi_int_reg_write_msk(dev,
				AD413X_REG_CHN(ch_nb),
				AD413X_ENABLE_M,
				AD413X_ENABLE_M);
else
       ret = ad413x_spi_int_reg_write_msk(dev,
				AD413X_REG_CHN(ch_nb),
				0x0U,
				AD413X_ENABLE_M);
if (ret)
			return ret;

```

Reply:
Even more condensation:
`ret = ad413x_spi_int_reg_write_msk(dev, AD413X_REG_CHN(ch_nb), enable ? AD413X_ENABLE_M : 0x0U, AD413X_ENABLE_M);`